### DISCUSSION TITLE
handleDataRequest not work

### DISCUSSION THREAD

**Discussion by Yiniau** [remix-run/remix#7595] [Q&A]
export handleDataRequest function in `app/entry.server.tsx`
```ts
export function handleDataRequest(
  response: Response,
  { request, params, context }: DataFunctionArgs,
) {
  console.log("request", request);

  const cors_headers = cors(request);
  for (const [k, v] of Object.entries(cors_headers)) {
    response.headers.set(k, v);
  }
  return response;
}
```
but not see any log in shell

<img width="1045" alt="image" src="https://github.com/remix-run/remix/assets/17947783/d9f89cfb-4f5d-4f51-b8de-0beb1d514c55">

**Comment by sergiodxa** [comment]
handleDataRequest only works when Remix does a client-side request to a UI route loader or action, which happens when it has the `?_data` search param.

If `/api_gateway/lemmy` is a resource route it will not be used, for resource routes you will have to apply CORS manually.

**Reply by mikeguo99** [reply]
This is very helpful, thank you! However, is this mentioned in the official documentation?
